atlasdb-client/src/main/java/com/palantir/atlasdb/util/TaggedMetricsInvocationEventHandler.java

```diff
@@ -72,12 +106,54 @@ public void onSuccess(@Nullable InvocationContext context, @Nullable Object resu
             return;
         }
 
+        if (result != null && ListenableFuture.class.isAssignableFrom(context.getMethod().getReturnType())) {
+            Futures.addCallback((ListenableFuture<?>) result, new FutureCallback<Object>() {
```

Comment:
It looks like you are casting `result` from `Object` to `ListenableFuture<?>`; can you do this without a check? (It looks to me like the check above is checking `context`'s return type, but I don't know if that is guaranteed to be the same as the type of `result`).

Reply:
they should be equivalent, I suppose I wanted to adhere to the interface in question as opposed to whatever type we may get back, I can add another check to the object itself.